### ISSUE TITLE
eth carried with the deposit erc20 token

### ISSUE THREAD

**Issue by liaoyinglong** [OffchainLabs/arbitrum-sdk#465]
**First of all, this is an awesome library!**

I have a question about when I call `Erc20Bridger.deposit` I can control the amount of `ETH` to send?
if not, how can I know I will send the `ETH` amount in `L1` and will receive the `ETH` amount in `L2` before do deposit action?

the code like:
```js
const bridger = new Erc20Bridger(l2Network)
// I want to know how much is  will send ETH amount in L1 and will receive the ETH amount in L2

const ethDepositTxResponse = await bridger.deposit({
  amount: utils.parseEther('10'),
  l1Signer: l1Signer /** <-- connected ethers-js Wallet */,
  l2Provider: l2Provider /** <--- ethers-js Provider */,
})

```

**Thank of all!**

**Comment by spsjvc** [comment]
For depositing ETH, you should use `EthBridger`. `Erc20Bridger` is used for depositing ERC-20 tokens.

**Comment by liaoyinglong** [comment]
yes, I mean when I deposit `ERC20 token` that will send some `ETH` to bridge in this line 
https://github.com/OffchainLabs/arbitrum-sdk/blob/01af6ef017e164bb7cb7febf7026ce60af76a112/src/lib/assetBridger/erc20Bridger.ts#L696

and can I know how much `eth` amount will be sent to my L2 address ?

**Comment by spsjvc** [comment]
The ETH sent is there to cover for the [retryable](https://docs.arbitrum.io/arbos/#retryables) execution fees and it's necessary for your ERC-20 deposit to succeed.

You can calculate this amount upfront by using the `L1ToL2MessageGasEstimator`. You can see here how the `Erc20Bridger` uses it: https://github.com/OffchainLabs/arbitrum-sdk/blob/main/src/lib/assetBridger/erc20Bridger.ts#L686

**Comment by liaoyinglong** [comment]
thank you!
The amount is send to bridge's `ETH` amount, can i get the amount of will received in `L2` address?

example:
- when i deposit `ERC20` token to `L2`
- send to bridge's `ETH` is `0.00004740420016352 ETH` in `L1`
- and I received `ETH` is `0.0000395035 Ether` in `L2` from the Smart Contract

I'd like to know if it's possible to calculate this value  `0.0000395035 Ether` before making a deposit.

<img width="869" alt="SCR-20240426-orqc" src="https://github.com/OffchainLabs/arbitrum-sdk/assets/23382699/575f1742-997b-4b50-973a-ab4eb3dc84dc">
<img width="908" alt="SCR-20240426-osym" src="https://github.com/OffchainLabs/arbitrum-sdk/assets/23382699/fe8b82c8-a6e1-4f5e-9b83-bb1e6a00cc92">
